Maid 1386 - Chunk store shall not panic

hintofbasil opened the pull request:
Hi,
I hope I'm not jumping the gun by attempting this before the sprint has officially begun.
I know my code introduces two new unwraps which I probably should handle but I was unsure on how to properly handle an error being returned.
I also checked for more panicable code but could only see the two functions.  If there are any I missed I can update them.
Any other comments would be highly welcome!

William

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/maidsafe/safe_vault/275)
<!-- Reviewable:end -->

maidsafe-highfive commented:
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @Fraser999 (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

hintofbasil commented:
Hello,
I've got a failing Travis test on the Linux build however when I run the specific test on my computer (also Linux) it passes fine.
Could I possibly get some insight into the issue as it's very hard to debug without being able to replicate?

dirvine commented:
Could be the latest nightly (runs on CI) has updated some part. If you make sure you are updated and do a `cargo update` then you should see the same issue locally I would hope. I am working on another issue just now so have not yet looked at the code. Some of the team will soon I imagine, if not I will get to it as soon as I can. 
Ah update, on checking the CI it could be a routing issue. This sprint will sort connections in routing and this looks like you have got this error in CI, I will rerun it and see, but it's not your code AFAIK.

dirvine commented on src/error.rs:
```diff
@@ -0,0 +1,44 @@
+impl ::std::error::Error for ChunkStoreError {
+    fn description(&self) -> &str {
+        match *self {
+            ChunkStoreError::DirectoryCreate => "Failed to create directory",
```
Could maybe use `From` defined for io error to allow more granular resolution of the error (disk fail/full/corrupt/permissions etc.)

hintofbasil commented:
Glad to hear the CI issues are not on my end.
I've updated to implement From.
I don't know if you would like the implementation of Display to be more verbose.  Also any advice on the two unwraps I've added?  I didn't want to force other constructor functions to return Result.

Fraser999 commented:
Thanks for this PR.  I've merged it now, so if you could let us know your bitcoin wallet address we'll get you paid up shortly.

hintofbasil commented:
Glad it was accepted!
My address is 12H1LRC1esAcwVTK8M7GLvT7hDcUWjtV97.

NickLambert commented:
Payment on the way, thanks for your help!
